### PR TITLE
Sebastian Sloan

### PR DESCRIPTION
**Pull request by sasloan**
I completed user story 3 in time. and I was going to try the extension however I decided I wanted to do an active record query for the list of chef ingredients instead of the nested enum in the view. So I spent the last bit of my time playing with that instead of attempting the extension.

### CODE REVIEW

File: app/controllers/chef_ingredients_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class ChefIngredientsController < ApplicationController
+
+	def index
+		@chef = Chef.find(params[:chef_id])
+		@dishes = @chef.dishes
```
Comment (megstang) [inline app/controllers/chef_ingredients_controller.rb]: You don't need to send this instance variable over.. everything that's in dishes is accessible from the @ chef variable.

File: app/models/dish.rb [hunk 2]
```diff
@@ -1,4 +1,11 @@
 class Dish <ApplicationRecord
   validates_presence_of :name, :description
   belongs_to :chef
+
+	has_many :dish_ingredients
+	has_many :ingredients, through: :dish_ingredients
+
+	def total_calories
+		ingredients.sum(:calories)
```
Comment (megstang) [inline app/models/dish.rb]: 👍

File: app/views/chef_ingredients/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<h2> Chef Ingredients </h2>
+
+<% @dishes.each do |dish| %>
+	<% dish.ingredients.each do |ingredient| %>
+		<p><%= ingredient.name %></p>
+	<% end %>
+<% end %>
```
Comment (megstang) [inline app/views/chef_ingredients/index.html.erb]: The way you're printing this out is not going to be a unique list of ingredients if a chef has multiple dishes with the same ingredients used. This is a place where you should use an AR query to get a list of unique ingredients for a chef and iterate through that and display those ingredients.

File: app/views/dishes/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<h2> <%= @dish.name %></h2>
+
+<h3> Chef: <%= @dish.chef.name %></h3>
+
+<% @dish.ingredients.each do |ingredient| %>
+<h4> Ingredients </h4>
+		<p><%= ingredient.name %> :
+			<%= ingredient.calories %> calories </p>
+	</section>
```
Comment (megstang) [inline app/views/dishes/show.html.erb]: Not sure why you have this section here

File: app/views/dishes/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<h2> <%= @dish.name %></h2>
+
+<h3> Chef: <%= @dish.chef.name %></h3>
+
+<% @dish.ingredients.each do |ingredient| %>
+<h4> Ingredients </h4>
```
Comment (megstang) [inline app/views/dishes/show.html.erb]: This header is going to get printed out every time you print out an ingredient.. take it out of the iteration block

**Comment by sasloan** [comment]
Meg.


Yeah I was working on the method but I couldn't get the joins to work right
to I settled for something instead of nothing. I literally figured it out
10 min. After turning it in and user story 4 as well. I get it, just get a
little panicked when I'm almost out of time. Something I am working on.

**Pull request closed without merging**